pkg/storage/swift/swift.go

```diff
@@ -87,13 +212,34 @@ func (d *driver) RemoveStorage(cr *imageregistryv1.Config) (bool, error) {
 		return false, nil
 	}
 
-	return false, nil
+	client, err := d.getSwiftClient(cr)
+	if err != nil {
+		return false, err
+	}
+
+	_, err = containers.Delete(client, cr.Spec.Storage.Swift.Container).Extract()
+	if err != nil {
+		util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionUnknown, err.Error(), err.Error())
+		return false, err
+	}
+
+	cr.Spec.Storage.Swift.Container = ""
+	d.Config.Container = ""
+
+	if !reflect.DeepEqual(cr.Status.Storage.Swift, d.Config) {
+		cr.Status.Storage.Swift = d.Config.DeepCopy()
+	}
+
+	util.UpdateCondition(cr, imageregistryv1.StorageExists, operatorapi.ConditionFalse, "Swift Container Deleted", "The swift container has been removed.")
+
+	return true, nil
 }
 
 func (d *driver) Volumes() ([]corev1.Volume, []corev1.VolumeMount, error) {
 	return nil, nil, nil
 }
 
 func (d *driver) CompleteConfiguration(cr *imageregistryv1.Config) error {
+	cr.Status.Storage.Swift = d.Config.DeepCopy()
```

Comment:
@dmage can you please weigh in on what this function is supposed to do in the storage driver's lifecycle?
Comparing the `s3`, `pvc`, and `emptyDir` implementations I see very different things going on.

Reply:
It can be just `return nil`. When this PR is merged, I'll create a follow-up PR to remove this function from the interface, we already have all defaultings in CreateStorage.

Reply:
thanks, I deleted this line since it's not necessary.